cmake/XrplSanitizers.cmake

```diff
@@ -0,0 +1,152 @@
+#[===================================================================[
+   Configure sanitizers based on environment variables.
+
+   This module reads the following environment variables:
+   - SANITIZERS: The sanitizers to enable. Possible values:
+     - "Address"
+     - "Address,UndefinedBehavior"
+     - "Thread"
+     - "Thread,UndefinedBehavior"
+
+   The compiler type and platform are detected automatically by CMake.
+   The sanitizer compile options are applied to the 'common' interface library
+   which is linked to all targets in the project.
+#]===================================================================]
+
+# Read environment variable
+set(SANITIZERS $ENV{SANITIZERS})
+
+if(NOT SANITIZERS)
+  return()
+endif()
+
+message(STATUS "Configuring sanitizers: ${SANITIZERS}")
+
+# Parse SANITIZERS value to determine which sanitizers to enable
+set(ENABLE_ASAN FALSE)
+set(ENABLE_TSAN FALSE)
+set(ENABLE_UBSAN FALSE)
+
+if(SANITIZERS MATCHES "Address")
+    set(ENABLE_ASAN TRUE)
+endif()
+if(SANITIZERS MATCHES "Thread")
+    set(ENABLE_TSAN TRUE)
+endif()
+if(SANITIZERS MATCHES "UndefinedBehavior")
+    set(ENABLE_UBSAN TRUE)
+endif()
```

Comment:
ideally, you should chop off the matched part
and in the end check `SANITIZERS` is empty

this will prevent issues like setting incorrect value/typo and so on